README.md

```diff
@@ -1,4 +1,681 @@
+  Use the user and api_key properties of the above objects as the username and password. Place these values into the Authorization header of the HTTP request in the form Basic <base64 encoding of username and password joined by a single colon (:)>.
+
+- To authenticate using a bearer token:
+  To obtain your token using the IBM Cloud CLI, first log in to IBM Cloud, then run the following command:
+  ```
+    ibmcloud iam oauth-tokens
+  ```
+    Place this token in the Authorization header of the HTTP request in the form Bearer. Both API key or JWT tokens are supported.
+
+- To authenticate directly using the api_key:
+  Place the key directly as the value of the X-Auth-Token HTTP header.
+
+```golang
+    URL := os.Getenv("ADMIN_ENDPOINT")
+	apiKey := os.Getenv("API_KEY")
+
+	if URL == "" {
+		fmt.Println("Please set env ADMIN_ENDPOINT")
+		os.Exit(1)
+	}
+
+	if apiKey == "" {
+		fmt.Println("Please set env API_KEY")
+		os.Exit(1)
+	}
+
+	basicAuthenticator, err := core.NewBasicAuthenticator("token", apiKey)
+	if err != nil {
+		fmt.Printf("Failed to Authenticate with error : %s\n", err.Error())
+		os.Exit(1)
+	}
+```
+
+
+### Create the service
+
+Create a new service object.
+
+```golang
+    // Create a new REST service.
+	ibmEventStreamsAdminRestApiService, serviceErr := ibmeventstreamsadminrestapiv1.NewIbmEventStreamsAdminRestApiV1(&ibmeventstreamsadminrestapiv1.IbmEventStreamsAdminRestApiV1Options{
+		URL:           URL,
+		Authenticator: basicAuthenticator,
+	})
+	if serviceErr != nil {
+		fmt.Println("Error Creating Service")
+		os.Exit(1)
+	}
+```
+
+### Creating a Kafka topic
+
+To get a Kafka topic detail information, the admin REST SDK issues a GET request to the `/admin/topics/TOPICNAME`
+path (where `TOPICNAME` is the name of the Kafka topic that you want to get).  
+
+Expected status codes
+  - 200: Retrieve topic details successfully in following format:
+
+
+#### Example
+
+```golang
+	// Set the retries policy.
+	serviceAPI.EnableRetries(0, 0)
+
+	// Construct an instance of the createTopicOptionsModel.
+	createTopicOptionsModel := new(ibmeventstreamsadminrestapiv1.CreateTopicOptions)
+	createTopicOptionsModel.Name = core.StringPtr("test-topic")
+	createTopicOptionsModel.Partitions = core.Int64Ptr(int64(26))
+	createTopicOptionsModel.PartitionCount = core.Int64Ptr(int64(1))
+	createTopicOptionsModel.Headers = map[string]string{"x-custom-header": "x-custom-value"}
+
+	// Create the Topic.
+	response, operationErr := serviceAPI.CreateTopic(createTopicOptionsModel)
+	if operationErr != nil {
+		return fmt.Errorf("Error Creating Topics: %s\n", operationErr.Error())
+	}
+
+	// Check the result.
+	if response.StatusCode != http.StatusAccepted {
+		return fmt.Errorf("Error Creating Topic: status %d\n", response.StatusCode)
+	}
+
+	fmt.Printf("\tname: %s created\n", *createTopicOptionsModel.Name) 
+```
+
+### Deleting a Kafka topic
+
+To delete a Kafka topic, the admin REST SDK issues a DELETE request to the `/admin/topics/TOPICNAME`
```

Comment:
I'm not sure if we need the REST API description in all these sections. This SDK encapsulate the REST API so I'm not sure users are interested to know what HTTP method and body are required. The swagger is available if needed

Reply:
Not sure I want to change this as the README comes from the Admin Rest project and is used everywhere for generating not just this readme, but the apidocs as well.